File: PixelDungeonPhotoMaker/src/au/net/genesis/mds/particles/FloatUp.java
package au.net.genesis.mds.particles;


public class FloatUp extends ParticleLogic {

	
	@Override
	public void initialise() {
		super.initialise();
		yspeed = -1;
		alpha = 0;
		xscale = yscale =1F;
	}

	@Override
	public void update() {
		super.update();
		if (age > lifeTime - 10) {
			alpha = Math.max(alpha-.1F, 0F);
		} else {
			alpha = Math.min(alpha+.2F, 1F);
		}
	}
}

File: src/main/java/com/mdsimmo/particles/FloatUp.kt
package com.mdsimmo.particles

class FloatUp : ParticleLogic() {
    override fun newParticle(): ParticleState {
        return ParticleState(
            yspeed = -1f,
            alpha = 0f
        )
    }

    override fun update(particle: ParticleState): ParticleState {
        return super.update(particle).copy(
                alpha = if (particle.age > particle.lifeTime - 10) {
                    (particle.alpha - .1f).coerceAtLeast(0f)
                } else {
                    (particle.alpha + .2f).coerceAtMost(1f)
                }
        )
    }
}